pkg/satellite/api.go

```diff
@@ -0,0 +1,121 @@
+package satellite
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/graphql-go/graphql"
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/auth"
+	"storj.io/storj/pkg/satellite/satelliteql"
+)
+
+const (
+	authorization = "Authorization"
+	contentType   = "Content-Type"
+
+	authorizationBearer = "Bearer"
+
+	applicationJSON    = "application/json"
+	applicationGraphql = "application/graphql"
+)
+
+func (gw *gateway) grapqlHandler(w http.ResponseWriter, req *http.Request) {
+	token := getToken(req)
+	//TODO: handle error
+	query, _ := getQuery(req)
```

Comment:
Yup, handle error, if proper handling requires too much code, you can do:
```
if err != nil {
    // TODO: better handling
    http.Error(...)
    return
}
```

This would at l